sympy/physics/mechanics/joint.py

```diff
@@ -526,9 +526,9 @@ def _set_angular_velocity(self):
 
     def _set_linear_velocity(self):
         self.parent_point.set_vel(self.parent.frame, 0)
-        self.child_point.set_vel(self.parent.frame, 0)
+        self.child_point.set_vel(self.child.frame, 0)
```

Comment:
In the pin joint, wouldn't the velocity of both the parent and child points be zero in both frames?

Reply:
Yes, but this way I thought that the `vel` method gets a bit easier job searching and defining both will probably give cycles.

Reply:
Setting this velocity to `0` may actually not be correct at all, see #23393. As example you may actually use a `PinJoint` in combination with a `PrismaticJoint`.

Reply:
What's the precise example that demonstrates this?

Also, I'm wondering if we can only really set these velocities once the whole system of body connections is defined. Maybe it is ill designed thinking we can set velocities correctly as we go along.

Reply:
```py
from sympy.physics.mechanics import Body, PinJoint, PrismaticJoint, dynamicsymbols

q1, q2, u1, u2 = dynamicsymbols('q1, q2, u1, u2')

wall = Body('N')
block = Body('A')
pend = Body('B')

slider = PrismaticJoint('J1', wall, block, coordinates=q1, speeds=u1)
hinge = PinJoint('J2', block, pend, coordinates=q2, speeds=u2)

for point in (block.masscenter, slider.parent_point, slider.child_point):
    print(point)
    print(point.vel(wall.frame))
    print(point.vel(block.frame), end='\n\n')
```
These should all be the same.

Reply:
The only change is that the velocity should not be set to `0`, but use the `masscenter` as reference.

Reply:
I feel like we are missing something. Could it be the case that the different `ReferenceFrames` can have a different zero-velocity reference? Because than the entire part about the `PrismaticJoint` isn't an issue.